File: Easy/Predict the Column.py
#User function Template for python3

class Solution:
    def columnWithMaxZeros(self, arr, N):
        # code here 

        m = -1
        ans = -1
        
        ok = False
        
        for i in range(N):
            
            temp = 0
            
            for j in range(N):
                if arr[j][i] == 0:
                    temp += 1
                    ok = True
                
                if temp > m:
                    ans = i
                    m = temp
                
        if ok:
            return ans
        
        return -1


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        arr = []
        for i in range(N):
            line = [int(x) for x in input().strip().split()]
            arr.append(line)
        ob=Solution()
        print(ob.columnWithMaxZeros(arr,N))


# } Driver Code Ends

File: Easy/Predict the Column.cpp
//{ Driver Code Starts
//Initial Template for C++
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++

class Solution{
    public:
    /*Function to count zeros in each column
    * N : Number of rows and columns in array
    M is the matrix that is globally declared
    */
    int columnWithMaxZeros(vector<vector<int>>arr,int N)
    {
        
        // Your code here
        int ans = -1;
        int m = INT_MIN;
        for (int i=0; i<N; i++)
        {
            int colonna = 0;
            for (int j=0; j<N; j++)
            {
                if (arr[j][i] == 0) colonna++;
            }
            if (colonna > m)
            {
                m = colonna;
                ans = i;
            }
            
        }
        
        
        return ans;
    }
};

//{ Driver Code Starts.
int main() {
	
	int testcase;
	
	cin >> testcase;
	
	while(testcase--){
	    int N;
	    cin >> N;
	    vector<vector<int>>arr(N,vector<int>(N));
	    // Taking elements input into the matrix
	    for(int i = 0;i<N;i++){
	        for(int j = 0;j<N;j++){
	            int x;
	            cin >> x; 
	            arr[i][j]=x;
	        }
	    }
	    Solution ob;
	    cout << ob.columnWithMaxZeros(arr,N) << endl;
	}
	
	return 0;
}
// } Driver Code Ends